tasks/DependencyCheck/task/index.ts

```diff
@@ -17,65 +20,73 @@ async function run(): Promise<void> {
     tl.debug(`Setting resource path to ${taskManifestPath}`);
     tl.setResourcePath(taskManifestPath);
 
-    const workspaceId: string = tl.getInput('workspaceId', true) as string;
-    const sharedKey: string = tl.getInput('sharedKey', true) as string;
-    const enableSelfHostedDatabase: boolean = tl.getBoolInput('enableSelfHostedDatabase', true);
-    const databaseEndpoint: string = tl.getInput('databaseEndpoint', (enableSelfHostedDatabase)) as string;
-    const scanPath: string = tl.getInput('scanPath', true) as string;
-    const excludedScanPathPatterns: string = tl.getInput('excludedScanPathPatterns', true) as string;
-
-    let repositoryName = (tl.getVariable('Build.Repository.Name'))?.split('/')[1];
-    let branchName = tl.getVariable('Build.SourceBranchName');
-    let buildName = tl.getVariable('Build.DefinitionName');
-    let buildNumber = tl.getVariable('Build.BuildNumber');
-    let commitId = tl.getVariable('Build.SourceVersion');
-
-    const la: ILogAnalyticsClient = new LogAnalyticsClient(
-      workspaceId,
-      sharedKey,
-    );
+    const enableVulnerabilityFilesMaintenance: boolean = tl.getBoolInput('enableVulnerabilityFilesMaintenance', true);
+    const writeStorageAccountContainerSasUri: string = tl.getInput('writeStorageAccountContainerSasUri', false) as string;
+    const workspaceId: string = tl.getInput('workspaceId', false) as string;
+    const sharedKey: string = tl.getInput('sharedKey', false) as string;
+    const enableSelfHostedVulnerabilityFiles: boolean = tl.getBoolInput('enableSelfHostedVulnerabilityFiles', false);
+    const readStorageAccountContainerSasUri: string = tl.getInput('readStorageAccountContainerSasUri', false) as string;
+    const scanPath: string = tl.getInput('scanPath', false) as string;
+    const excludedScanPathPatterns: string = tl.getInput('excludedScanPathPatterns', false) as string;
 
     const scriptBasePath = `${__dirname}/dependency-check-cli/bin/dependency-check`;
     const scriptFullPath = process.platform === 'win32' ? `${scriptBasePath}.bat` : `${scriptBasePath}.sh`;
+
     tl.debug(`Dependency check script path set to ${scriptFullPath}`);
 
     if (!(process.platform === 'win32')) {
       spawnSync('chmod', ['+x', scriptFullPath])
     }
 
-    if (enableSelfHostedDatabase) {
-      const trimmedDatabaseEndpoint = databaseEndpoint.replace(/\/$/, '');
-      cleanDependencyCheckData();
-      await getVulnData(trimmedDatabaseEndpoint, 'odc.mv.db', `${__dirname}/dependency-check-cli/data/odc.mv.db`);
-      await getVulnData(trimmedDatabaseEndpoint, 'jsrepository.json', `${__dirname}/dependency-check-cli/data/jsrepository.json`);
-    }
+    if (!enableVulnerabilityFilesMaintenance) {
+      let repositoryName = (tl.getVariable('Build.Repository.Name'))?.split('/')[1];
```

Comment:
This assumes that Build.Repository.Name is always formatted the same.  There are different providers, may be worth checking the value of Build.Repository.Provider before setting this variable

Reply:
Great point, when Azure DevOps repo (TfsGit) it is just the name of the repository instead of GitHub's <org>/<repository name>. Added a check of the Build.Repository.Provider value when setting repositoryName.